topological_navigation/scripts/localisation.py

```diff
@@ -153,15 +158,18 @@ def MapCallback(self, msg) :
 
     def Callback(self, msg, item):
         #print item
-        val = getattr(msg, item['field'])
-        if val == item['val'] :
-            if item['name'] not in self.loc_by_topic:
-                self.loc_by_topic.append(item['name'])
-        else:
-            if item['name'] in self.loc_by_topic:
-                self.loc_by_topic.remove(item['name'])
-
-
+        dist = get_distance(self.current_pose, self.previous_pose)
+        if dist>0.10:
+            val = getattr(msg, item['field'])
+            if val == item['val'] :
+                if item['name'] not in self.loc_by_topic:
+                    self.loc_by_topic.append(item['name'])
+                    self.previous_pose = self.current_pose
+            else:
+                if item['name'] in self.loc_by_topic:
+                    self.loc_by_topic.remove(item['name'])
+                    self.previous_pose = self.current_pose
+                    self.previous_pose.position.x = self.previous_pose.position.x+1000
```

Comment:
Yes this is because previous_pose is the position were the topic changed value this could be less than 10 cms away from a position where the topic will be read so I need to set it to a position that will imply constant checking, this can only cause problems if there is another localise by topic node 1000 (+/- 0.1) meters away on the x axis from another one. As I said I will find a more elegant solution afterwards but this is the way I managed to make it work given the urgent need for this